src/main/kotlin/de/bund/digitalservice/useid/identification/IdentificationSessionService.kt

```diff
@@ -42,21 +54,83 @@ class IdentificationSessionService(private val identificationSessionRepository:
         return "${applicationProperties.baseUrl}$IDENTIFICATION_SESSIONS_BASE_PATH/${session.useIdSessionId}/$TCTOKEN_PATH_SUFFIX"
     }
 
+    @Transactional
     fun startSessionWithEIdServer(useIdSessionId: UUID): TCTokenType {
-        val identificationSession = identificationSessionRepository.findByUseIdSessionId(useIdSessionId)
-            ?: throw IdentificationSessionNotFoundException(useIdSessionId)
-        val tcToken: TCTokenType
+        val requestData: RequestData = getRequestData(useIdSessionId)
+
+        val tcTokenWrapper: TcTokenWrapper?
         try {
-            val eidService = EidService(eidServiceConfig, identificationSession.requestDataGroups)
-            tcToken = eidService.getTcToken("${applicationProperties.baseUrl}$REFRESH_PATH")
+            tcTokenWrapper = soapHandler.getTcToken(requestData, "${applicationProperties.baseUrl}$REFRESH_PATH")
```

Comment:
I am wondering whether we could rename the soapHandler to something more specific: eIdSoapHandler maybe?